falcon/util/uri.py

```diff
@@ -311,7 +314,7 @@ def parse_query_string(query_string, keep_blank_qs_values=False):
                     # will be the same for 'foo=1,,3' as 'foo=1&foo=&foo=3'.
                     v = [element for element in v if element]
```

Comment:
Individual elements in an array induced by "," should be decoded IMHO.  It's just recursive design logic.  Why would one type of value be encoded/decoded, whereas another would not?  This is counter-intuitive and makes no sense.  Values are encoded so that we can pass the funny characters without having issues.  Just go to each element in the array and decode it.

I know it sounds un-fast.  But first make it correct, then make it fast.